File: test_data/code/python/1897-redistribute-characters-to-make-all-strings-equal.py
Python
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        if not words:
            return True

        total_chars = {}
        for word in words:
            for char in word:
                if char not in total_chars:
                    total_chars[char] = 0
                total_chars[char] += 1

        freq = []
        for count in total_chars.values():
            if count % len(words) > 0:
                return False
            freq.append(count // len(words))

        return all(count % len(words) == 0 for count in total_chars.values())

File: test_data/code/java/1897-redistribute-characters-to-make-all-strings-equal.java
class Solution {
    public boolean makeEqual(String[] words) {
        int[] charCount = new int[26];
        int totalChars = 0;

        // Count the total number of characters and count the occurrences of each character
        for (String word : words) {
            int[] wordCount = new int[26];
            for (char c : word.toCharArray()) {
                wordCount[c - 'a']++;
            }
            for (int i = 0; i < 26; i++) {
                charCount[i] += wordCount[i];
            }
            totalChars += word.length();
        }

        // Check if all strings can be made equal
        for (int count : charCount) {
            if (count % (totalChars / words.length) != 0) {
                return false;
            }
        }
        return true;
    }
}